### DISCUSSION TITLE
opencv4.1.0 do_install failed

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#243] [Q&A] [answered]
Hi,
I´s using repo: dunfell. A month ago I compiled the code successfully. 
Now I´m trying to reproduce the process on another laptop and it failed on the last step, the opencv can not be installed sccessfully.
does someone have idea about this error? thanks in advanced.

Loading cache: 100% |############################################################################################################################################| Time: 0:00:00
Loaded 3623 entries from dependency cache.
Parsing recipes: 100% |##########################################################################################################################################| Time: 0:00:02
Parsing of 2448 .bb files complete (2446 cached, 2 parsed). 3625 targets, 114 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.46.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-oe4t-linux"
MACHINE              = "jetson-nano-devkit-emmc"
DISTRO               = "tegrademo"
DISTRO_VERSION       = "3.1.21+snapshot-20230105"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 = "HEAD:aa0073041806c9f417a33b0b7f747d2a86289eda"
meta-tegra
contrib              = "HEAD:1dd1910d7533789837bf910e52a2afb4b1dbfe31"
meta-oe
meta-python
meta-networking
meta-filesystems     = "HEAD:7203130ed8b58c0df75cb72222ac2bcf546bce44"
meta-virtualization  = "HEAD:beea119eb529b4a11f266004aee8b548427aea39"
meta-tegra-community = "HEAD:4920b6a4386aa64bcb8db449d05fbd234acdc861"
meta-tegra-support
meta-demo-ci
meta-tegrademo
workspace            = "dunfell:bf6e5da15ad923712298456739fe5aab2b0057f7"
meta-testapp        = "HEAD:139f8d7448614695a56ac14f1e3a9ad2a49cff15"

Initialising tasks: 100% |#######################################################################################################################################| Time: 0:00:05
Sstate summary: Wanted 21 Found 0 Missed 21 Current 3770 (0% match, 99% complete)
NOTE: Executing Tasks

**### ERROR: opencv-4.1.0-r0 do_install: Execution of '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/temp/run.do_install.4128843' failed with exit code 1
ERROR: Logfile of failure stored in: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/temp/log.do_install.4128843**

Log data follows:
| DEBUG: Executing python function extend_recipe_sysroot
| NOTE: Direct dependencies are ['/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-multimedia/v4l2apps/v4l-utils_1.18.1.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/contrib/recipes-multimedia/gstreamer-1.14/gstreamer1.0-plugins-base_1.14.4.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-multimedia/ffmpeg/ffmpeg_4.2.2.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/python-numpy/python3-numpy_1.17.4.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-graphics/jpeg/libjpeg-turbo_2.0.4.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-core/glibc/glibc_2.31.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-multimedia/libpng/libpng_1.6.37.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-multimedia/webp/libwebp_1.1.0.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/opencv/ade_0.1.1f.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-cross_8.3.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/pkgconfig/pkgconfig_git.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/recipes-devtools/cuda/cuda-libraries_10.2.460-1.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/ninja/ninja_1.10.0.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-extended/bzip2/bzip2_1.0.8.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/recipes-devtools/cuda/cuda-cudart_10.2.300-1.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-graphics/gphoto2/libgphoto2_2.5.25.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-multimedia/libtiff/tiff_4.1.0.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/tbb/tbb_2020.2.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/patch/patch_2.7.6.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/swig/swig_3.0.12.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/libtool/libtool_2.4.6.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/pseudo/pseudo_git.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/quilt/quilt-native_0.66.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-core/zlib/zlib_1.2.11.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/python/python3_3.8.14.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-core/glib-2.0/glib-2.0_2.62.6.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/recipes-devtools/cmake/tegra-cmake-overrides_1.0.3.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/contrib/recipes-multimedia/gstreamer-1.14/gstreamer1.0_1.14.4.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/libeigen/libeigen_3.3.7.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/recipes-devtools/cuda/cuda-compiler_10.2.460-1.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/glog/glog_0.3.5.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/gflags/gflags_2.2.2.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-tegra/contrib/recipes-devtools/gcc-8/gcc-runtime_8.3.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/cmake/cmake-native_3.16.5.bb:do_populate_sysroot', 'virtual:native:/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-devtools/python/python3_3.8.14.bb:do_populate_sysroot', '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta/recipes-gnome/gtk+/gtk+3_3.24.14.bb:do_populate_sysroot']
| NOTE: Installed into sysroot: []
| NOTE: Skipping as already exists in sysroot: ['v4l-utils', 'gstreamer1.0-plugins-base', 'ffmpeg', 'python3-numpy', 'libjpeg-turbo', 'glibc', 'libpng', 'libwebp', 'ade', 'gcc-cross-aarch64', 'pkgconfig-native', 'cuda-libraries', 'ninja-native', 'bzip2', 'cuda-cudart-native', 'libgphoto2', 'tiff', 'tbb', 'patch-native', 'swig-native', 'libtool', 'pseudo-native', 'quilt-native', 'zlib', 'python3', 'glib-2.0', 'tegra-cmake-overrides', 'gstreamer1.0', 'libeigen', 'cuda-compiler-native', 'glog', 'gflags', 'gcc-runtime', 'cmake-native', 'python3-native', 'gtk+3', 'systemd', 'libx11', 'libxrender', 'libtheora', 'pango', 'util-linux', 'libvorbis', 'gobject-introspection', 'orc', 'alsa-lib', 'wayland', 'libogg', 'libdrm', 'iso-codes', 'libxv', 'libsm', 'wayland-protocols', 'libglvnd', 'xz', 'x264', 'libxcb', 'linux-libc-headers', 'texinfo-dummy-native', 'xz-native', 'flex-native', 'libmpc-native', 'automake-native', 'autoconf-native', 'libtool-native', 'gmp-native', 'zlib-native', 'gnu-config-native', 'binutils-cross-aarch64', 'mpfr-native', 'cuda-cudart', 'cuda-nvrtc', 'libcusolver', 'cuda-nvgraph', 'libcufft', 'libcusparse', 'libcublas', 'libnpp', 'libcurand', 're2c-native', 'opkg-utils', 'tar-l4t-workaround-native', 'zstd-l4t-workaround-native', 'cuda-driver-native', 'cuda-nvcc-headers-native', 'libxdamage', 'libxkbcommon', 'at-spi2-atk', 'fontconfig', 'cairo', 'gdk-pixbuf', 'libxfixes', 'atk', 'libepoxy', 'libxrandr', 'libxext', 'libxcursor', 'libxcomposite', 'libxi', 'libexif', 'libusb-compat', 'libxml2', 'attr-native', 'libpcre-native', 'readline', 'gdbm', 'sqlite3', 'libxcrypt', 'openssl', 'libtirpc', 'libnsl2', 'autoconf-archive', 'libffi', 'libpcre', 'dbus', 'bash-completion', 'libcap', 'cuda-nvprune-native', 'cuda-nvcc-native', 'cuda-cuobjdump-native', 'libunwind', 'ncurses-native', 'expat-native', 'bzip2-native', 'curl-native', 'openssl-native', 'util-linux-native', 'gdbm-native', 'libnsl2-native', 'readline-native', 'libffi-native', 'libtirpc-native', 'sqlite3-native', 'libgcc', 'shadow', 'base-passwd', 'shadow-sysroot', 'base-files', 'kmod', 'acl', 'shadow-native', 'libpam', 'xtrans', 'xorgproto', 'util-macros', 'fribidi', 'harfbuzz', 'libxft', 'freetype', 'libcap-ng', 'ncurses', 'expat', 'libpciaccess', 'libpthread-stubs', 'libice', 'l4t-nvidia-glheaders', 'libxdmcp', 'libxau', 'xcb-proto', 'gettext-minimal-native', 'm4-native', 'cuda-nvcc-headers', 'cuda-driver', 'xkeyboard-config', 'at-spi2-core', 'pixman', 'shared-mime-info', 'libusb1', 'libatomic-ops', 'libpcre2-native', 'libcap-ng-native', 'attr', 'cracklib', 'flex', 'icu', 'tegra-libraries-cuda', 'libxtst', 'tegra-libraries-core']
| DEBUG: Python function extend_recipe_sysroot finished
| DEBUG: Executing shell function do_install
| NOTE: DESTDIR=/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/image VERBOSE=1 cmake --build /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/build --target install --
| [0/1] /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot-native/usr/bin/cmake -S/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/git -B/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/build
| Re-run cmake no build system arguments
| CMake Warning at cmake/OpenCVUtils.cmake:656 (message):
|   Unexpected option: WITH_EIGEN (=ON)
|
|   Condition: IF (NOT;WINRT;AND;NOT;CMAKE_CROSSCOMPILING)
| Call Stack (most recent call first):
|   CMakeLists.txt:255 (OCV_OPTION)
|
|
| -- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) (Required is at least version "2.7")
| -- Looking for ccache - not found
| -- Found ZLIB: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/lib/libz.so (found suitable version "1.2.11", minimum required is "1.2.3")
| -- Found ZLIB: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/lib/libz.so (found version "1.2.11")
| -- Found TBB (env): /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/lib/libtbb.so
| -- CUDA detected: 10.2
| -- Automatic detection of CUDA generation failed. Going to build for all known architectures.
| -- CUDA NVCC target flags: --gpu-architecture=compute_53 --gpu-code=sm_53  --include-path /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/local/cuda-10.2/include --library-path /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/local/cuda-10.2/lib;-gencode;arch=compute_53,code=sm_53;-D_FORCE_INLINES
| -- Could not find OpenBLAS include. Turning OpenBLAS_FOUND off
| -- Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off
| -- Could NOT find Atlas (missing: Atlas_CBLAS_INCLUDE_DIR Atlas_CLAPACK_INCLUDE_DIR Atlas_CBLAS_LIBRARY Atlas_BLAS_LIBRARY Atlas_LAPACK_LIBRARY)
| -- Could NOT find BLAS (missing: BLAS_LIBRARIES)
| -- LAPACK requires BLAS
| -- A library with LAPACK API not found. Please specify library location.
| -- OpenCV Python: during development append to PYTHONPATH: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/build/python_loader
| -- Caffe:   NO
| -- Protobuf:   NO
| -- Glog:   YES
| -- freetype2:   YES
| -- harfbuzz:    YES
| -- Module opencv_ovis disabled because OGRE3D was not found
| -- No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
| -- Found installed version of gflags: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot/usr/lib/cmake/gflags
| -- Detected gflags version: 2.2.2
| -- Checking SFM deps... TRUE
| -- CERES support is disabled. Ceres Solver for reconstruction API is required.
| -- Module opencv_dnn_objdetect disabled because opencv_dnn dependency can't be resolved!
| abort()ing pseudo client by server request. See https://wiki.yoctoproject.org/wiki/Pseudo_Abort for more details on this.
| Check logfile: /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/pseudo//pseudo.log
| Aborted (core dumped)
| FAILED: build.ninja
| /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/recipe-sysroot-native/usr/bin/cmake -S/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/git -B/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/build
| ninja: error: rebuilding 'build.ninja': subcommand failed
| WARNING: exit code 1 from a shell command.
| ERROR: Execution of '/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/build/tmp/work/armv8a_tegra210-oe4t-linux/opencv/4.1.0-r0/temp/run.do_install.4128843' failed with exit code 1
**### ERROR: Task (/home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/opencv/opencv_4.1.0.bb:do_install) failed with exit code '1'**
NOTE: Tasks Summary: Attempted 6817 tasks of which 6816 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/test/Develop/Nvidia_Yocto/tegra-demo-distro/layers/meta-oe/recipes-support/opencv/opencv_4.1.0.bb:do_install
Summary: There was 1 ERROR message shown, returning a non-zero exit code.

**Comment by madisongh** [comment, marked as answer]
This is a problem with the upstream recipe in meta-oe; it downloads some of the sources to a directory not under `${S}`, but doesn't tell `pseudo` about it.  We have a bbappend in the kirkstone and later branches that fixes the problem; if you're using an older branch, try copying [this bbappend](https://github.com/OE4T/tegra-demo-distro/blob/master/layers/meta-tegrademo/recipes-support/opencv/opencv_%25.bbappend).   The problem only affects rebuilds of opencv in a workspace where it's already been through the compile/install steps.  You can work around it by `bitbake -c clean opencv` then building again.

I can probably drop this from the master branch, since [this commit](https://git.openembedded.org/meta-openembedded/commit/meta-oe/recipes-support/opencv?id=6b7cf95e75cbfc08cf1f6eddc7ce3f6a0a39edcf) upstream relocated the `contrib` repo under `${S}`; that was for other reasons, but coincidentally also fixes this issue.

**Reply by danny1209** [reply]
Hi @madisongh 
thanks a lot!
Yes I´m doing rebuild.
`bitbake -c clean opencv` and then rebuild. it solves my problem!
